tests/FakeItEasy.Tests/ServiceLocatorTests.cs

```diff
@@ -13,17 +13,25 @@ namespace FakeItEasy.Tests
     public class ServiceLocatorTests
     {
         [SuppressMessage("Microsoft.Performance", "CA1823:AvoidUnusedPrivateFields", Justification = "Used reflectively.")]
-        private readonly Type[] singletonTypes =
+        private static IEnumerable<Type> SingletonTypes
         {
-            typeof(IExpressionCallMatcherFactory),
-            typeof(ExpressionArgumentConstraintFactory),
-            typeof(IProxyGenerator)
-        };
+            get
+            {
+                yield return typeof(IExpressionCallMatcherFactory);
+                yield return typeof(ExpressionArgumentConstraintFactory);
+                yield return typeof(IProxyGenerator);
+            }
+        }

```

Comment:
> I can change it to an array if you prefer.

Sure, if you have spare cycles, but as I say, it's not hugely important here. More of a conversation point.